packages/terra-application-links/src/tabs/_TabMenu.jsx

```diff
@@ -0,0 +1,168 @@
+  /**
+   * Should the menu be hidden, set to true if there are no hidden items.
+   */
+  isHidden: PropTypes.bool,
+  /**
+   * Ref callback for menu button.
+   */
+  refCallback: PropTypes.func,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class TabMenu extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.handleChildClick = this.handleChildClick.bind(this);
+    this.handleOnRequestClose = this.handleOnRequestClose.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.getTargetRef = this.getTargetRef.bind(this);
+    this.setTargetRef = this.setTargetRef.bind(this);
+    this.wrapOnClick = this.wrapOnClick.bind(this);
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  getTargetRef() {
+    return this.targetRef;
+  }
+
+  setTargetRef(node) {
+    this.targetRef = node;
+
+    if (this.props.refCallback) {
+      this.props.refCallback(node);
+    }
+  }
+
+  handleOnRequestClose() {
+    this.setState({ isOpen: false });
+  }
+
+  handleOnClick() {
+    this.setState({ isOpen: true });
+  }
+
+  handleOnKeyDown(event) {
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.ENTER) {
+      this.setState({ isOpen: true });
+    }
+  }
+
+  wrapOnClick(child) {
+    return (event) => {
+      if (child.props.onClick) {
+        child.props.onClick(event);
+      }
+
+      this.setState({ isOpen: false });
+    };
+  }
+
+  createRoutes(popup) {
+    const props = {
+      role: 'tab',
+      tabIndex: '0',
+      onClick: this.handleOnClick,
+      onKeyDown: this.handleOnKeyDown,
+      'data-application-tabs-more': true,
+    };
+
+    const routes = this.props.children.map(child => (
+      <Route
+        path={child.props.path}
+        key={child.props.path}
+        render={() => (
+          <TabMenuDisplay
+            {...props}
+            text={child.props.text}
+            popup={popup}
+            refCallback={this.setTargetRef}
+            isSelected
+            isHidden={this.props.isHidden}
+          />
+        )}
+      />
+    ));
+
+    const { intl } = this.context;
+    const menuToggleText = intl.formatMessage({ id: 'Terra.application.tabs.more' });
+    routes.push(
+      <Route
+        key={'menu-display-more'}
+        render={() => (
+          <TabMenuDisplay
+            {...props}
+            text={menuToggleText}
+            popup={popup}
+            refCallback={this.setTargetRef}
+            isHidden={this.props.isHidden}
+          />
+        )}
+      />,
+    );
+
+    return routes;
+  }
+
+  handleChildClick() {
+    if (this.state.isOpen) {
+      this.setState({ isOpen: false });
+    }
+  }
+
+  createHiddenTabs() {
+    return (
+      <TabMenuList>
```

Comment:
@dkasper-was-taken in regards to regaining focus, do you think we apply a approach similar to how select manages it? https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/_SelectMenu.jsx#L43

Reply:
I have an updated implementation just checked in.

Reply:
Nice!